qiskit/opflow/gradients/derivative_base.py

```diff
@@ -130,14 +136,19 @@ def parameter_expression_grad(param_expr: ParameterExpression,
         if param not in param_expr._parameter_symbols:
             return 0.0
 
-        import sympy as sy
         expr = param_expr._symbol_expr
         keys = param_expr._parameter_symbols[param]
-        expr_grad = sy.N(0)
         if not isinstance(keys, IterableAbc):
             keys = [keys]
-        for key in keys:
-            expr_grad += sy.Derivative(expr, key).doit()
+        if HAS_SYMENGINE:
```

Comment:
Is this still needed or can it be put inside ParameterExpression?

Reply:
I'm not sure, I don't really know a lot about the gradients code. I just did it here to adapt what was being done to work with a `ParameterExpression` that could either be symengine or sympy based.

Maybe @Cryoris  or @Zoufalc can provide a better answer.

Reply:
@BryceFuller could you have a look at this?

Reply:
I'm quite sure it can be removed since we have this as method in the `ParameterExpression`. We have to check that the code uses the `ParameterExpression.gradient` method instead of this function.

Reply:
@Cryoris I'm just suggesting replacing the `if HAS_SYMENGINE` clause with 
```python
expr_grad += param_expr.gradient(expr, key)
```
So that opflow doesn't have to worry about whether to use symengine or sympy. ParameterExpression can worry about that.

Reply:
Yeah, but we should remove that code anyways, it's been on my list since a long time 😉 With removing it we essentially delegate all gradients to ParameterExpression, so opflow won't need any `HAS_SYMENGINE` as you said above.

Reply:
With #6360 being merged this no longer is necessary and I removed this in the recent rebase.